// react-login-signup-form/lib/loginSignupForm.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _loader = _interopRequireDefault(require("./loader.js"));

var _loginForm = _interopRequireDefault(require("./loginForm.js"));

var _signupForm = _interopRequireDefault(require("./signupForm.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } _setPrototypeOf(subClass.prototype, superClass && superClass.prototype); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.getPrototypeOf || function _getPrototypeOf(o) { return o.__proto__; }; return _getPrototypeOf(o); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var backgroundStyle = {
  position: 'fixed',
  width: '100%',
  height: '100%',
  backgroundColor: '#3498DB'
};
var containerStyle = {
  position: 'absolute',
  top: '50%',
  left: '50%',
  backgroundColor: '#2E86C1D8',
  padding: '3rem 2rem',
  'msTransform': 'translateX(-50%) translateY(-50%)',
  'webkitTransform': 'translate(-50%,-50%)',
  'transform': 'translate(-50%,-50%)',
  borderTopLeftRadius: '1rem'
};

var LoginSignupForm =
/*#__PURE__*/
function (_React$Component) {
  function LoginSignupForm(props) {
    var _this;

    _classCallCheck(this, LoginSignupForm);

    _this = _possibleConstructorReturn(this, _getPrototypeOf(LoginSignupForm).call(this, props));

    _defineProperty(_defineProperty(_defineProperty(_defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "_login", function (username, password) {
      _this.props.dispatch(_this.props.tryLoginAction(username, password));
    }), "_openLoginForm", function () {
      _this.setState({
        showSignupForm: false
      });
    }), "_signup", function (email, password) {
      _this.props.dispatch(_this.props.trySignupAction(email, password));
    }), "_openSignupForm", function () {
      _this.setState({
        showSignupForm: true
      });
    });

    _this.state = {
      showSignupForm: false
    };
    return _this;
  }

  _createClass(LoginSignupForm, [{
    key: "componentWillMount",
    value: function componentWillMount() {
      this._updateStyles();
    }
  }, {
    key: "componentWillReceiveProps",
    value: function componentWillReceiveProps(nextProps) {
      if (this.props.isLoggedIn !== nextProps.isLoggedIn && nextProps.isLoggedIn) {
        this.props.history.replace(this.props.redirectUrl);
      }
    }
  }, {
    key: "_updateStyles",
    value: function _updateStyles() {
      // Background Styles
      if (this.props.backgroundStyle !== null) {
        backgroundStyle = this.props.backgroundStyle;
      } else {
        if (this.props.backgroundColor !== null) {
          backgroundStyle.backgroundColor = this.props.backgroundColor;
        }

        if (this.props.backgroundImageUrl !== null) {
          backgroundStyle.backgroundImage = "url('".concat(this.props.backgroundImageUrl, "')");
        }

        if (this.props.backgroundRepeat !== null) {
          backgroundStyle.backgroundRepeat = this.props.backgroundRepeat ? 'repeat' : 'no-repeat';
        }

        if (this.props.backgroundSize !== null) {
          backgroundStyle.backgroundSize = this.props.backgroundSize;
        }
      } // Login Container Box Styles


      if (this.props.containerStyle !== null) {
        containerStyle = this.props.containerStyle;
      } else {
        if (this.props.containerColor !== null) {
          containerStyle.backgroundColor = this.props.containerColor;
        }

        if (this.props.containerPosition !== null) {
          if (this.props.containerPosition <= 1 && this.props.containerPosition >= 0) {
            containerStyle.left = "".concat(this.props.containerPosition * 100, "%");
          }
        }
      }
    }
  }, {
    key: "render",
    value: function render() {
      return _react.default.createElement("div", null, _react.default.createElement(_loader.default, {
        loading: this.props.isLoading
      }), _react.default.createElement("div", {
        style: backgroundStyle
      }, _react.default.createElement("div", {
        style: containerStyle
      }, !this.state.showSignupForm ? _react.default.createElement(_loginForm.default, {
        loginButton: this.props.loginButton,
        errorFontColor: this.props.errorFontColor,
        errorFontSize: this.props.errorFontSize,
        errorFontStyle: this.props.errorFontStyle,
        errorStyle: this.props.errorStyle,
        inputBorderColor: this.props.inputBorderColor,
        inputBorderFocusedColor: this.props.inputBorderFocusedColor,
        inputElement: this.props.inputElement,
        inputFontColor: this.props.inputFontColor,
        inputHintColor: this.props.inputHintColor,
        inputHintFocusedColor: this.props.inputHintFocusedColor,
        inputWidth: this.props.inputWidth,
        login: this._login,
        openSignupForm: this._openSignupForm,
        signupButton: this.props.signupButton
      }) : _react.default.createElement(_signupForm.default, {
        loginButton: this.props.loginButton,
        errorFontColor: this.props.errorFontColor,
        errorFontSize: this.props.errorFontSize,
        errorFontStyle: this.props.errorFontStyle,
        errorStyle: this.props.errorStyle,
        inputBorderColor: this.props.inputBorderColor,
        inputBorderFocusedColor: this.props.inputBorderFocusedColor,
        inputElement: this.props.inputElement,
        inputFontColor: this.props.inputFontColor,
        inputHintColor: this.props.inputHintColor,
        inputHintFocusedColor: this.props.inputHintFocusedColor,
        inputWidth: this.props.inputWidth,
        openLoginForm: this._openLoginForm,
        signup: this._signup,
        signupButton: this.props.signupButton
      }))));
    }
  }]);

  _inherits(LoginSignupForm, _React$Component);

  return LoginSignupForm;
}(_react.default.Component);

LoginSignupForm.propTypes = {
  backgroundColor: _propTypes.default.string,
  backgroundImageUrl: _propTypes.default.string,
  backgroundRepeat: _propTypes.default.bool,
  backgroundSize: _propTypes.default.string,
  backgroundStyle: _propTypes.default.object,
  loginButton: _propTypes.default.node,
  containerColor: _propTypes.default.string,
  containerPosition: _propTypes.default.number,
  containerStyle: _propTypes.default.object,
  errorFontColor: _propTypes.default.string,
  errorFontSize: _propTypes.default.string,
  errorFontStyle: _propTypes.default.string,
  errorStyle: _propTypes.default.object,
  inputBorderColor: _propTypes.default.string,
  inputBorderFocusedColor: _propTypes.default.string,
  inputElement: _propTypes.default.node,
  inputFontColor: _propTypes.default.string,
  inputHintColor: _propTypes.default.string,
  inputHintFocusedColor: _propTypes.default.string,
  inputWidth: _propTypes.default.string,
  isLoading: _propTypes.default.bool.isRequired,
  isLoggedIn: _propTypes.default.bool.isRequired,
  redirectUrl: _propTypes.default.string.isRequired,
  signupButton: _propTypes.default.node,
  tryLoginAction: _propTypes.default.func.isRequired,
  trySignupAction: _propTypes.default.func.isRequired
};
LoginSignupForm.defaultProps = {
  backgroundColor: null,
  backgroundImageUrl: null,
  backgroundRepeat: null,
  backgroundSize: null,
  backgroundStyle: null,
  loginButton: null,
  containerColor: null,
  containerPosition: null,
  containerStyle: null,
  errorFontColor: null,
  errorFontSize: null,
  errorFontStyle: null,
  errorStyle: null,
  inputBorderColor: null,
  inputBorderFocusedColor: null,
  inputElement: null,
  inputFontColor: null,
  inputHintColor: null,
  inputHintFocusedColor: null,
  inputWidth: null,
  signupButton: null
};
var _default = LoginSignupForm;
exports.default = _default;
