// react-login-signup-form/lib/signupForm.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _input = _interopRequireDefault(require("./input.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } _setPrototypeOf(subClass.prototype, superClass && superClass.prototype); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _getPrototypeOf(o) { _getPrototypeOf = Object.getPrototypeOf || function _getPrototypeOf(o) { return o.__proto__; }; return _getPrototypeOf(o); }

var buttonStyle = {
  float: 'right',
  cursor: 'pointer',
  marginTop: '1rem',
  padding: '0.5rem 1rem',
  color: '#FFF',
  fontSize: '0.875rem',
  backgroundPosition: 'center',
  transition: 'background 0.8s'
};
var errorStyle = {
  fontSize: '0.875rem',
  height: '0.875rem',
  color: '#CC2C21',
  fontStyle: 'italic'
};

var SignupForm =
/*#__PURE__*/
function (_React$Component) {
  function SignupForm(props) {
    var _this;

    _classCallCheck(this, SignupForm);

    _this = _possibleConstructorReturn(this, _getPrototypeOf(SignupForm).call(this, props));
    _this.state = {
      email: '',
      password: '',
      confirmPassword: '',
      emailError: '',
      passwordError: '',
      confirmPasswordError: ''
    };
    return _this;
  }

  _createClass(SignupForm, [{
    key: "componentWillMount",
    value: function componentWillMount() {
      this._updateStyles();
    } // From https://stackoverflow.com/questions/46155/how-to-validate-an-email-address-in-javascript?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa

  }, {
    key: "_isEmailValid",
    value: function _isEmailValid(email) {
      var re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
      return re.test(String(email).toLowerCase());
    }
  }, {
    key: "_resetError",
    value: function _resetError() {
      this.setState({
        emailError: '',
        passwordError: '',
        confirmPasswordError: ''
      });
    }
  }, {
    key: "_isInputValid",
    value: function _isInputValid() {
      this._resetError();

      var isValid = true;

      if (this.state.email === '') {
        this.setState({
          emailError: 'Email cannot be empty'
        });
        isValid = false;
      } else if (!this._isEmailValid(this.state.email)) {
        this.setState({
          emailError: 'Email not valid'
        });
        isValid = false;
      }

      if (this.state.password === '') {
        this.setState({
          passwordError: 'Password cannot be empty'
        });
        isValid = false;
      }

      if (this.state.confirmPassword === '') {
        this.setState({
          confirmPasswordError: 'Password cannot be empty'
        });
        isValid = false;
      } else if (this.state.confirmPassword !== this.state.password) {
        this.setState({
          confirmPasswordError: 'Passwords must match'
        });
        isValid = false;
      }

      return isValid;
    }
  }, {
    key: "_openLoginForm",
    value: function _openLoginForm() {
      this.props.openLoginForm();
    }
  }, {
    key: "_signup",
    value: function _signup() {
      if (this._isInputValid()) {
        var _this$state = this.state,
            email = _this$state.email,
            password = _this$state.password;
        this.props.signup(email, password);
      }
    }
  }, {
    key: "_onEmailChange",
    value: function _onEmailChange(val) {
      this.setState({
        email: val
      });
    }
  }, {
    key: "onPasswordChange",
    value: function onPasswordChange(val) {
      this.setState({
        password: val
      });
    }
  }, {
    key: "onConfirmPasswordChange",
    value: function onConfirmPasswordChange(val) {
      this.setState({
        confirmPassword: val
      });
    }
  }, {
    key: "_updateStyles",
    value: function _updateStyles() {
      // Button Styles
      if (this.props.buttonColor !== null) {
        buttonStyle.color = this.props.buttonColor;
      } // Error message Styles


      if (this.props.errorStyle !== null) {
        errorStyle = this.props.errorStyle;
        errorStyle.height = this.props.errorStyle.fontSize || '1rem';
      } else {
        if (this.props.errorFontColor !== null) {
          errorStyle.color = this.props.errorFontColor;
        }

        if (this.props.errorFontSize !== null) {
          errorStyle.fontSize = this.props.errorFontSize;
          errorStyle.height = this.props.errorFontSize;
        }

        if (this.props.errorFontStyle !== null) {
          errorStyle.fontStyle = this.props.errorFontStyle;
        }
      }
    }
  }, {
    key: "_renderEmailInput",
    value: function _renderEmailInput() {
      var _this2 = this;

      return _react.default.createElement("div", null, _react.default.cloneElement(this.props.inputElement, {
        onChange: function onChange(val) {
          return _this2._onEmailChange(val);
        },
        value: this.state.email
      }), _react.default.createElement("div", {
        style: errorStyle
      }, this.state.emailError));
    }
  }, {
    key: "_renderPasswordInput",
    value: function _renderPasswordInput() {
      var _this3 = this;

      return _react.default.createElement("div", null, _react.default.cloneElement(this.props.inputElement, {
        onChange: function onChange(val) {
          return _this3.onPasswordChange(val);
        },
        value: this.state.password,
        isPassword: true
      }), _react.default.createElement("div", {
        style: errorStyle
      }, this.state.passwordError));
    }
  }, {
    key: "_renderConfirmPasswordInput",
    value: function _renderConfirmPasswordInput() {
      var _this4 = this;

      return _react.default.createElement("div", null, _react.default.cloneElement(this.props.inputElement, {
        onChange: function onChange(val) {
          return _this4.onConfirmPasswordChange(val);
        },
        value: this.state.confirmPassword,
        isPassword: true
      }), _react.default.createElement("div", {
        style: errorStyle
      }, this.state.confirmPasswordError));
    }
  }, {
    key: "_renderLoginButton",
    value: function _renderLoginButton() {
      var _this5 = this;

      var style = {
        float: 'left',
        marginTop: '1rem'
      };
      return _react.default.createElement("div", {
        style: style
      }, _react.default.cloneElement(this.props.loginButton, {
        onClick: function onClick() {
          return _this5._openLoginForm();
        }
      }));
    }
  }, {
    key: "_renderSignupButton",
    value: function _renderSignupButton() {
      var _this6 = this;

      var style = {
        float: 'right',
        marginTop: '1rem'
      };
      return _react.default.createElement("div", {
        style: style
      }, _react.default.cloneElement(this.props.signupButton, {
        onClick: function onClick() {
          return _this6._signup();
        }
      }));
    }
  }, {
    key: "render",
    value: function render() {
      var _this7 = this;

      return _react.default.createElement("div", null, this.props.inputElement ? _react.default.createElement("div", null, this._renderEmailInput(), this._renderPasswordInput(), this._renderConfirmPasswordInput()) : _react.default.createElement("div", null, _react.default.createElement(_input.default, {
        borderColor: this.props.inputBorderColor,
        borderFocusedColor: this.props.inputBorderFocusedColor,
        fontColor: this.props.inputFontColor,
        hintColor: this.props.inputHintColor,
        hintFocusedColor: this.props.inputHintFocusedColor,
        hintText: "Email",
        onChange: function onChange(val) {
          return _this7._onEmailChange(val);
        },
        value: this.state.email,
        width: this.props.inputWidth
      }), _react.default.createElement("div", {
        style: errorStyle
      }, this.state.emailError), _react.default.createElement(_input.default, {
        borderColor: this.props.inputBorderColor,
        borderFocusedColor: this.props.inputBorderFocusedColor,
        fontColor: this.props.inputFontColor,
        hintColor: this.props.inputHintColor,
        hintFocusedColor: this.props.inputHintFocusedColor,
        hintText: "Password",
        isPassword: true,
        onChange: function onChange(val) {
          return _this7.onPasswordChange(val);
        },
        value: this.state.password,
        width: this.props.inputWidth
      }), _react.default.createElement("div", {
        style: errorStyle
      }, this.state.passwordError), _react.default.createElement(_input.default, {
        borderColor: this.props.inputBorderColor,
        borderFocusedColor: this.props.inputBorderFocusedColor,
        fontColor: this.props.inputFontColor,
        hintColor: this.props.inputHintColor,
        hintFocusedColor: this.props.inputHintFocusedColor,
        hintText: "Confirm Password",
        isPassword: true,
        onChange: function onChange(val) {
          return _this7.onConfirmPasswordChange(val);
        },
        value: this.state.confirmPassword,
        width: this.props.inputWidth
      }), _react.default.createElement("div", {
        style: errorStyle
      }, this.state.confirmPasswordError)), this.props.loginButton ? this._renderLoginButton() : _react.default.createElement("div", {
        onClick: function onClick() {
          return _this7._openLoginForm();
        },
        style: _objectSpread({}, buttonStyle, {
          float: 'left'
        })
      }, "LOGIN"), this.props.signupButton ? this._renderSignupButton() : _react.default.createElement("div", {
        onClick: function onClick() {
          return _this7._openSignupForm();
        },
        style: buttonStyle
      }, "SIGNUP"));
    }
  }]);

  _inherits(SignupForm, _React$Component);

  return SignupForm;
}(_react.default.Component);

SignupForm.propTypes = {
  loginButton: _propTypes.default.node,
  errorFontColor: _propTypes.default.string,
  errorFontSize: _propTypes.default.string,
  errorFontStyle: _propTypes.default.string,
  errorStyle: _propTypes.default.object,
  inputBorderColor: _propTypes.default.string,
  inputBorderFocusedColor: _propTypes.default.string,
  inputElement: _propTypes.default.node,
  inputFontColor: _propTypes.default.string,
  inputHintColor: _propTypes.default.string,
  inputHintFocusedColor: _propTypes.default.string,
  inputWidth: _propTypes.default.string,
  openLoginForm: _propTypes.default.func.isRequired,
  signup: _propTypes.default.func.isRequired,
  signupButton: _propTypes.default.node
};
SignupForm.defaultProps = {
  loginButton: null,
  errorFontColor: null,
  errorFontSize: null,
  errorFontStyle: null,
  errorStyle: null,
  inputBorderColor: null,
  inputBorderFocusedColor: null,
  inputElement: null,
  inputFontColor: null,
  inputHintColor: null,
  inputHintFocusedColor: null,
  inputWidth: null,
  signupButton: null
};
var _default = SignupForm;
exports.default = _default;
